packages/lesswrong/server/repos/DialogueChecksRepo.ts

```diff
@@ -58,6 +63,32 @@ class DialogueChecksRepo extends AbstractRepo<"DialogueChecks"> {
       );
     `, [userId1, userId2])
   }
+
+  async getMatchFormYourTurn(): Promise<DialogueCheckWithExtraData[]> {
+    const result = await this.getRawDb().any(`
+      SELECT 
+        dc.*,
+        dmp_reciprocal._id AS "targetUserMatchPreferenceId"
+      FROM "DialogueChecks" AS dc
+      LEFT JOIN "DialogueMatchPreferences" AS dmp ON dc._id = dmp."dialogueCheckId"
+      LEFT JOIN
+          "DialogueChecks" AS dc_reciprocal
+          ON
+              dc."userId" = dc_reciprocal."targetUserId"
+              AND dc."targetUserId" = dc_reciprocal."userId"
+      LEFT JOIN "DialogueMatchPreferences" AS dmp_reciprocal ON dc_reciprocal._id = dmp_reciprocal."dialogueCheckId"
+      LEFT JOIN "Notifications" AS n ON n."extraData"->>'targetUserMatchPreferenceId' = dmp_reciprocal._id
```

Comment:
I am also pretty unsure that the way this query is constructed is right, or will perform reasonably, given that `Notifications` is a big table and `extraData` isn't indexed at all.  If you do [this](https://github.com/ForumMagnum/ForumMagnum/pull/8548/files#r1433352642), you'll be able to join on `documentId`.

Separately, I don't see `targetUserMatchPreferenceId` being used anywhere in the code after being passed in to the notification?  If it's not needed, then my guess is that most of this query goes away (and maybe can actually just be an existing query that we already have for finding matched dialogue checks).

Reply:
In testing it runs instantly. (I build queries using Hex so tend to get exposed to them being slow)

`targetUserMatchPreferenceId` is used... wait for it... in the JOIN in this query itself! 
Because what we're doing is notifying users about a new form to pay attention to. And the id of that form is a pointer to the new ground truth object we're trying to notify them about.